tests/compiler/std/map.optimized.wat

```diff
@@ -3667,6 +3661,10 @@
   local.set $1
   loop $for-loop|1
    local.get $0
+   i32.const 24
+   i32.shl
+   i32.const 24
+   i32.shr_s
```

Comment:
This is a `for (let k: i8 = 0; k < 100; ++k)` loop detecting that incrementing an i8 can yield an unwrapped value, hence recompiling with unified wrap states since unaware of the `k < 100` limit. As such expected, but also an optimization opportunity.